react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    };
+    this.arrowClicks = this.arrowClicks.bind(this);
+    this.renderProgressDots = this.renderProgressDots.bind(this);
+    this.clickDots = this.clickDots.bind(this);
+  }
+
+  componentDidMount() {
+    setInterval(() => {
+      if (this.state.index === this.props.images.length - 1) {
+        this.setState({
+          index: 0
+        });
+      } else {
+        this.setState({
+          index: this.state.index + 1
+        });
+      }
+    }, 3000);
```

Comment:
Be sure to store the interval id so that you can halt/reset the interval when the user clicks on an arrow/dot.